Build failure: lha

## Steps To Reproduce

Steps to reproduce the behavior:

1. `NIXPKGS_ALLOW_UNFREE=1 nix build github:NixOS/nixpkgs/88195a94#lha --impure`

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
note: keeping build directory '/var/tmp/nix-build-lha-unstable-2021-01-07.drv-2'
error: builder for '/nix/store/31k2qff923l8z012i6ysz7lji7sy4ldx-lha-unstable-2021-01-07.drv' failed with exit code 2;
       last 25 log lines:
>       |                    long long int   uint64_t {aka long unsigned int}
>       |                 %ld
> header.c:213:29: warning: format '%llx' expects argument of type 'long long unsigned int', but argument 3 has type 'uint64_t' {aka 'long unsigned int'} [-Wformat=]
>   213 |         printf("%lld(%#016llx)\n", l, l);
>       |                      ~~~~~~~^         ~
>       |                             |         |
>       |                             |         uint64_t {aka long unsigned int}
>       |                             long long unsigned int
>       |                      %#016lx
> header.c: In function 'get_bytes':
> header.c:241:20: warning: format '%d' expects argument of type 'int', but argument 2 has type 'long int' [-Wformat=]
>   241 |         printf("%02d %2d: \"", get_ptr - start_ptr, len);
>       |                 ~~~^           ~~~~~~~~~~~~~~~~~~~
>       |                    |                   |
>       |                    int                 long int
>       |                 %02ld
> header.c: In function 'init_header':
> header.c:1568:9: warning: ignoring return value of 'readlink' declared with attribute 'warn_unused_result' [-Wunused-result]
>  1568 |         readlink(name, hdr->realname, sizeof(hdr->realname));
>       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> make[2]: *** [Makefile:424: header.o] Error 1
> make[2]: Leaving directory '/build/source/src'
> make[1]: *** [Makefile:373: all-recursive] Error 1
> make[1]: Leaving directory '/build/source'
> make: *** [Makefile:314: all] Error 2
       For full logs, run 'nix log /nix/store/31k2qff923l8z012i6ysz7lji7sy4ldx-lha-unstable-2021-01-07.drv'.
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.11-xanmod1, NixOS, 25.05 (Warbler), 25.05.20241213.3566ab7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/frontear/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/frontear/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/frontear/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/frontear/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/smghprxmwyp707kqljlm4fsxm7h5wd72-lix-2.91.1/share`
 - nixpkgs: `/nix/store/haa4yg8jy1pv2qdpai4ps0phyg7qpsnz-input-farm/nixpkgs
```

## Notify maintainers

@svanderburg

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/lh/lha/package.nix
+++ b/pkgs/by-name/lh/lha/package.nix
@@ -24,6 +24,7 @@ stdenv.mkDerivation {
     platforms = lib.platforms.unix;
     maintainers = with lib.maintainers; [
       sander
+      momeemt
     ];
     # Some of the original LHa code has been rewritten and the current author
     # considers adopting a "true" free and open source license for it.

--- a/pkgs/by-name/lh/lha/package.nix
+++ b/pkgs/by-name/lh/lha/package.nix
@@ -7,27 +7,30 @@
 
 stdenv.mkDerivation {
   pname = "lha";
-  version = "unstable-2021-01-07";
+  version = "1.14i-unstable-2024-11-27";
 
   src = fetchFromGitHub {
     owner = "jca02266";
     repo = "lha";
-    rev = "03475355bc6311f7f816ea9a88fb34a0029d975b";
-    sha256 = "18w2x0g5yq89yxkxh1fmb05lz4hw7a3b4jmkk95gvh11mwbbr5lm";
+    rev = "26b71be85a762098bdeb95f4533045c7dad86f31";
+    hash = "sha256-jiYTBqDXvxTdrvHYaK+1eo4xIpl+B9ZljhBBYD0BGzQ=";
   };
 
   nativeBuildInputs = [ autoreconfHook ];
 
-  meta = with lib; {
+  meta = {
     description = "LHa is an archiver and compressor using the LZSS and Huffman encoding compression algorithms";
-    platforms = platforms.unix;
-    maintainers = [ maintainers.sander ];
-    # Some of the original LhA code has been rewritten and the current author
+    homepage = "https://github.com/jca02266/lha";
+    platforms = lib.platforms.unix;
+    maintainers = with lib.maintainers; [
+      sander
+    ];
+    # Some of the original LHa code has been rewritten and the current author
     # considers adopting a "true" free and open source license for it.
     # However, old code is still covered by the original LHa license, which is
     # not a free software license (it has additional requirements on commercial
     # use).
-    license = licenses.unfree;
+    license = lib.licenses.unfree;
     mainProgram = "lha";
   };
 }

